presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1608,9 +1613,15 @@ else if (column.getExpression() instanceof Identifier) {
             return assignments.build();
         }
 
-        private void checkFunctionName(Statement node, QualifiedName functionName)
+        private void checkFunctionName(Statement node, QualifiedName functionName, boolean isTemporary)
         {
-            if (functionName.getParts().size() != 3) {
+            if (isTemporary && !functionName.getPrefix().equals(Optional.empty())) {
```

Comment:
What about `functionName.getParts().size() != 1`, which is identical but more explicit.

Reply:
Done